client/components/platform-checkout/save-user/checkout-page-save-user.js

```diff
@@ -40,28 +38,38 @@ const CheckoutPageSaveUser = () => {
 	return (
 		<>
 			<h3>Remember your details?</h3>
-			<CheckboxControl
-				checked={ isSaveDetailsChecked }
-				onChange={ setIsSaveDetailsChecked }
-				name="save_user_in_platform_checkout"
-				label={ __(
-					'Save my information for faster checkouts',
-					'woocommerce-payments'
-				) }
-				value="true"
-			/>
+			<span>
+				<label htmlFor="save_user_in_platform_checkout">
+					<input
+						type="checkbox"
+						checked={ isSaveDetailsChecked }
+						onChange={ () => {
+							setIsSaveDetailsChecked( ! isSaveDetailsChecked );
+						} }
```

Comment:
(minor) on a `useState` in these kinds of scenarios where the state would need to be toggled, a more appropriate way to update state would be via a functional update:
```
setIsSaveDetailsChecked( v => !v );
```

This is because _technically_, multiple state updates can be queued and processed in a batch. When they are processed, you want them to be processed in their correct order and with the correct state value. More on this: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

Reply:
After a brief misunderstanding - fixed in [a99015c](https://github.com/Automattic/woocommerce-payments/pull/4076/commits/a99015cd1a08dc9c5eae1b28faff8df8ba4de7d8)